mrcnn/model.py

```diff
@@ -2341,7 +2341,7 @@ def train(self, train_dataset, val_dataset, learning_rate, epochs, layers,
             keras.callbacks.TensorBoard(log_dir=self.log_dir,
                                         histogram_freq=0, write_graph=True, write_images=False),
             keras.callbacks.ModelCheckpoint(self.checkpoint_path,
-                                            verbose=0, save_weights_only=True),
+                                            verbose=0, save_weights_only=True, save_best=self.config.save_best_only),
```

Comment:
@akashdeepjassal I see that you renamed save_best=self.config.SAVE_BEST_ONLY into save_best=self.config.save_best_only, but I actually was talking about the ModelCheckpoin's kw parameter itself, it should now be keras.callbacks.ModelCheckpoint(..., save_best_only=self.config.save_best_only, ...)